pkg/server/services/addon.go

```
@@ -0,0 +1,113 @@
+			return fmt.Errorf("only support addon status update")
+		}
+
+		_, err := c.addonClient.AddonV1alpha1().ManagedClusterAddOns(addon.Namespace).UpdateStatus(ctx, addon, metav1.UpdateOptions{})
+		return err
+	default:
+		return fmt.Errorf("unsupported addon event action %s", eventType.Action)
+	}
+}
+
+func (c *AddonService) RegisterHandler(handler server.EventHandler) {
+	c.addonInformer.Informer().AddEventHandler(&cache.ResourceEventHandlerFuncs{
```

golangci-lint (1.64.8)

Error return value of `(k8s.io/client-go/tools/cache.SharedInformer).AddEventHandler` is not checked

(errcheck)